File: src/interfaces/favorite/favorite.dto.ts
export interface FavoriteDTO {
  id: number;
  productId: number;
  userId: number;
  savedDate: string;
  ebayItemId?: string;
  title?: string;
  price?: number;
  image?: string;
  url?: string;
  condition?: string;
  currency?: string;
  primaryCategoryId?: string;
}

File: src/main/java/com/smartcompare/favorite/domain/dto/FavoriteDTO.java
package com.smartcompare.favorite.domain.dto;

import lombok.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class FavoriteDTO {
    private Long id;
    private Long productId;
    private Long userId;
    private LocalDateTime savedDate;

    // Datos mínimos del producto eBay para guardar favoritos directamente
    private String ebayItemId;
    private String title;
    private Double price;
    private String image;
    private String url;
    private String condition;
    private String currency;
}
